vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/main/java/com/vaadin/flow/component/checkbox/CheckboxGroup.java

```diff
@@ -461,26 +459,21 @@ public Registration addSelectionListener(
 
     @Override
     public void onEnabledStateChanged(boolean enabled) {
-        if (isReadOnly()) {
-            setDisabled(true);
-        } else {
-            setDisabled(!enabled);
-        }
+        setDisabled(!enabled);
         getCheckboxItems().forEach(this::updateEnabled);
     }
 
     @Override
     public void setReadOnly(boolean readOnly) {
-        isReadOnly = readOnly;
+        getElement().setProperty("readonly", readOnly);
         if (isEnabled()) {
-            setDisabled(readOnly);
             refreshCheckboxes();
```

Comment:
Is this still required? Before the change, running this would disable individual checkboxes as well. After the change, individual checkboxes are not set as readonly. It looks like you can not tamper with the `CheckBoxGroup` value by making a checkbox writable on the client and changing it's value, as the `CheckBoxGroup` being readonly prevents its value from changing. So maybe marking individual checkboxes as readonly is not required.

So either we:
- Remove this part, at which point the override for `setReadOnly`, `getReadOnly` becomes unnecessary
- Or we update `refreshCheckboxes` to make individual checkboxes readonly as well